tools/pg_tools/macros.xml

```diff
@@ -0,0 +1,74 @@
+<?xml version="1.0"?>
+<macros>
+    <xml name="inputs_outputs">
+        <inputs>
+            <param format="tabular" help="" label="Input tabular dataset" name="in_file" type="data" />
+            <yield />
+        </inputs>
+        <outputs>
+            <data format="tabular" name="out_file" label="${tool.name} on ${on_string}" />
+        </outputs>
+    </xml>
+
+    <xml name="requirements">
+        <requirements>
+            <requirement type="package" version="@PG_VERSION@">postgresql</requirement>
+            <requirement type="package" version="0.1">pglite</requirement>
+        </requirements>
+    </xml>
+
+    <token name="@PG_VERSION@">9.6.3</token>
+
+    <token name="@HELP_FOOTER@">
+    The postgresql tools, are using a postgresql database in user-space. Databases are packed into a tarball and stored in the user-history.
+    </token>
+
+    <token name="@UNTAR_INFILE@"><![CDATA[
+        tar -xjvf '$infile' &&
+        export PWD=`pwd` &&
+        sed -i.bak "s|unix_socket_directories.*|unix_socket_directories = '\$PWD/postgresql/'|" ./postgresql/db/postgresql.conf
+    ]]>
+    </token>
+
+    <token name="@PG_SETUP@"><![CDATA[
+        pglite setup -d ./postgresql &&
+        @PG_START@
+    ]]>
+    </token>
+
+    <token name="@PSQL@"><![CDATA[
+        psql "\$(pglite url -d ./postgresql)"
+    ]]>
+    </token>
+
+    <token name="@STATUS_RUNNING@"><![CDATA[
+        timeout 30 bash -c 'until pglite status -d ./postgresql | grep -F -q "server is running"; do sleep 1; done' &&
```

Comment:
just fyi, the reason for my horribly long while loop was that [timeout is unavailable on OSX](https://stackoverflow.com/questions/3504945/timeout-command-on-mac-os-x). Not sure if we care about those guys (I don't really that much.)

Reply:
I don't care that much either and doubt that anyone one is running a server under OSX.
I think this is ready to go for a first version.